### ISSUE TITLE
block call hang forever

### ISSUE THREAD

**Issue by sunnycalsoft** [spring-projects/spring-framework#29216]
Call : this.reactiveMongoTemplate
                .updateFirst(query, update, Person.class).block()
Above call is from Scheduled annotated method and this call is hunged forever with below exception, not sure why this is happening?

"scheduling-1" #55 prio=5 os_prio=0 cpu=103.09ms elapsed=56346.14s tid=0x00007f29be0123d0 nid=0x3e waiting on condition  [0x00007f29ba5bf000]
   java.lang.Thread.State: WAITING (parking)
	at jdk.internal.misc.Unsafe.park(java.base@17.0.1/Native Method)
	- parking to wait for  <0x00000000bc5a91e8> (a java.util.concurrent.CountDownLatch$Sync)
	at java.util.concurrent.locks.LockSupport.park(java.base@17.0.1/Unknown Source)
	at java.util.concurrent.locks.AbstractQueuedSynchronizer.acquire(java.base@17.0.1/Unknown Source)
	at java.util.concurrent.locks.AbstractQueuedSynchronizer.acquireSharedInterruptibly(java.base@17.0.1/Unknown Source)
	at java.util.concurrent.CountDownLatch.await(java.base@17.0.1/Unknown Source)
	at reactor.core.publisher.BlockingSingleSubscriber.blockingGet(BlockingSingleSubscriber.java:87)
	at reactor.core.publisher.Mono.block(Mono.java:1707)

**Comment by bclozel** [comment]
This doesn't give us much to understand the problem. Also this class is not provided by Spring Framework.

Could you create a minimal application that reproduces the problem and create a new issue on the Spring Data MongoDB project? https://github.com/spring-projects/spring-data-mongodb/issues

Thanks
